### PR TITLE
enabled copy-all for programmatic fast-registration via FlyteRemote

### PR DESCRIPTION
**Pull request by granthamtaylor**
## Why are the changes needed?

One should be able to utilize `--copy-all` via `FlyteRemote.fast_register_workflow`

## What changes were proposed in this pull request?

I expose `--copy-all` to fast-registration.

### CODE REVIEW

File: flytekit/remote/remote.py [hunk 1]
```diff
@@ -874,6 +874,7 @@ def fast_register_workflow(
         version: typing.Optional[str] = None,
         default_launch_plan: typing.Optional[bool] = True,
         options: typing.Optional[Options] = None,
+        copy_all: bool = False,
```
Comment (thomasjpfan) [inline flytekit/remote/remote.py:877]: @wild-endeavor Since this is new API for `fast_register_workflow`, should we just expose `FastPackageOptions` here?

**Review by thomasjpfan — COMMENTED** [review]
Thank you for the PR! I left a question around `FastPackageOptions` for maintainers to decide which API to go with.

Context: `FastPackageOptions` covers all the features `copy_all` does and more.

File: flytekit/remote/remote.py (hunk 1 above)
Comment (wild-endeavor) [reply flytekit/remote/remote.py:877]: yup that sounds good. please do

Comment (granthamtaylor) [reply flytekit/remote/remote.py]: I have updated it and it works as expected. That being said, it is pretty rough compared to just `copy_all=True` or `--copy-all`

```python
from flytekit.tools.fast_registration import FastPackageOptions
remote.fast_register_workflow(
    my_workflow,
    fast_package_options=FastPackageOptions(ignores=[], copy_style="ALL")
)
```


```python
from flytekit.tools.fast_registration import FastPackageOptions
remote.fast_register_workflow(my_workflow, copy_all=True)
```

File: flytekit/remote/remote.py [hunk 2]
```diff
@@ -882,6 +883,7 @@ def fast_register_workflow(
         :param serialization_settings: The serialization settings to be used
         :param default_launch_plan: This should be true if a default launch plan should be created for the workflow
         :param options: Additional execution options that can be configured for the default launchplan
+        :param fast_package_options: Options to customize copy_all behavior
```
Comment (thomasjpfan) [inline flytekit/remote/remote.py:886]:
```suggestion
        :param fast_package_options: Options to customize copying behavior
```

**Review by thomasjpfan — APPROVED** [review]
Since this is passing a flag into another method, I am okay with not having a unit test for this.

**Review by wild-endeavor — APPROVED** [review]

File: flytekit/remote/remote.py (hunk 1 above)
Comment (wild-endeavor) [reply flytekit/remote/remote.py]: We can pre-create some of these objects and expose them as constants to improve the experience.

**Review by eapolinario — APPROVED** [review]

**Pull request merged**